### DISCUSSION TITLE
Realtional data duplicate

### DISCUSSION THREAD

**Discussion by sabikrahat** [pocketbase/pocketbase#3263] [Q&A]
I have a collection named `orders`  which have a field name `inventories` which indicates the `inventory` collection and take multiple collection. Everything is working fine. But I want to take same inventory 2 times. is it possible? If yes then how can I change the database? for better understand I have given some screenshots.

<img width="606" alt="image" src="https://github.com/pocketbase/pocketbase/assets/49031412/01960bac-0d72-4342-a61d-ccebb328008f">

here the  `Tesing Fabrics` I want to add again.

**Comment by ganigeorgiev** [comment]
You can't reference the same item in a single field multiple times.

This will depend on your use case and I don't have the full picture of it, but most likely you'll need to restructure your collections and create a new 3rd junction collection.

For example you could create collection "orderInventories" with 2 `relation` fields - one to the "inventories" and another one to the "orders" collection.
This way when you want to create an order, you can initiate an empty "orders" record and then use its id to create the related "orderInventories" records.
Optionally, the "orders" collection could also have a "status" field that will indicate in what state the order is. This flow could also allow users to continue a previous incomplete order.

To get an order with all its related inventories you could use [indirect/back relation expansion](https://pocketbase.io/docs/working-with-relations/#back-relation-expand), something like:

```js
const order = await pb.collection("orders").getOne("ORDER_ID", {
  expand: "orderInventories(order).inventory"
})
```

**Comment by sabikrahat** [comment]
hey @ganigeorgiev, I have found another way. Maybe it's more easier. I have taken a field named `inventoriesIds` where I'm storing the inventories ids though it is duplicated or not. While retrieving data, I'm searching the inventories from the `inventory` field. 

`
selectedInventories = order.inventoriesIds
          .map((e) => order.inventories.firstWhere((i) => i.id == e))
          .toList();
`
